pkg/types/types.go

```diff
@@ -234,40 +267,49 @@ func (s byStepTime) Less(i, j int) bool {
 	return s[i].StepTime < s[j].StepTime
 }
 
-func mergeMetrics(metrics []Metric) Metric {
+func mergeMetrics(metrics []Metric, mismatchCheck bool) (metric Metric, mismatches int) {
 	if len(metrics) == 0 {
-		return Metric{}
+		return Metric{}, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		m := metrics[0]
+		return m, 0
 	}
 
 	sort.Sort(byStepTime(metrics))
 	healed := 0
 
 	// metrics[0] has the highest resolution of metrics
-	metric := metrics[0]
+	metric = metrics[0]
 	for i := range metric.Values {
-		if !metric.IsAbsent[i] {
-			continue
-		}
-
-		// found a missing value, look for a replacement
+		pointExists := !metric.IsAbsent[i]
+		areValuesEqual := true
 		for j := 1; j < len(metrics); j++ {
+			if pointExists && !mismatchCheck {
```

Comment:
I think we can short-circuit here if `areValuesEqual == false` while in `mismatchCheck` mode
```suggestion
			if pointExists && !(mismatchCheck && areValuesEqual) {
```

Reply:
Again, it was something left out with future implementations in mind (i.e. majority voting) to see if we can bear the performance penalty or not. Anyways, will change it.